pithos/plugins/notification_icon.py

```diff
@@ -194,11 +194,10 @@ def song_changed(self, window, song):
             self.statusicon.set_tooltip_text("{} by {}".format(song.title, song.artist))
 
     def _toggle_visible(self, *args):
-        self.window.set_visible(not self.window.get_visible())
-
-        if self.window.get_visible(): # Ensure it's on top
+        if self.window.get_visible():
+            self.window.hide_on_delete()
```

Comment:
Just use `self.window.hide()`. That method is only a convenience thing for directly connecting signals.